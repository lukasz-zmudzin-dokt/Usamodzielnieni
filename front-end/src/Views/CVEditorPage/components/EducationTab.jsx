import React from "react";
import movie_2 from "../../../assets/movie_2.png";
import { CVEditorTab, ActionWithDate } from '.';

const EducationTab = ({ data, onChange, onPrevClick, onNextClick }) => (
    <CVEditorTab
        title="Edukacja"
        movie={movie_2}
        onPrevClick={onPrevClick}
        onNextClick={onNextClick}
    >
        <ActionWithDate data={data} onChange={onChange} />
    </CVEditorTab>
)

export default EducationTab;