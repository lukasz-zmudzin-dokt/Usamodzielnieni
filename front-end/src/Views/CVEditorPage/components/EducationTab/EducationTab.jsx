import React from "react";
import movie_2 from "assets/movie_2.png";
import { CVEditorTab, ActionWithDate } from "../";

const EducationTab = ({
  data,
  onChange,
  onPrevClick,
  onNextClick,
  comments,
  loading,
  showComments,
  validated,
  error,
  video,
  errVid,
  formTab,
}) => (
  <CVEditorTab
    title="Edukacja"
    movie={movie_2}
    video={video}
    onPrevClick={onPrevClick}
    onNextClick={onNextClick}
    comments={comments}
    loading={loading}
    error={error}
    showComments={showComments}
    errVid={errVid}
    formTab={formTab}
  >
    <ActionWithDate
      data={data}
      onChange={onChange}
      place={{ required: true, minLength: 1, maxLength: 200 }}
      description={{ required: true, minLength: 1, maxLength: 150 }}
      validated={validated}
      required
    />
  </CVEditorTab>
);

export default EducationTab;
