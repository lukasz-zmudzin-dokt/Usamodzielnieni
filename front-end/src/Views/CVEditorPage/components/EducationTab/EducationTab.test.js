import React from 'react';
import { render } from '@testing-library/react';
import EducationTab from './EducationTab';

describe('EducationTab', () => {
    let props;
    beforeEach(() => {
        props = {
            data: [], 
            onChange: () => {}
        }
    });

    it('should render without crashing', () => {
        const { container } = render(
            <EducationTab {...props} />
        );

        expect(container).toMatchSnapshot();
    });
});