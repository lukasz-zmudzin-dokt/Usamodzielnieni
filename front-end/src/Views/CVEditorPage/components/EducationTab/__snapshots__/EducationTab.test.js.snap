// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationTab should render without crashing 1`] = `
<div>
  <div>
    <h3>
      Edukacja
    </h3>
    <div
      class="form-group"
    >
      <img
        alt=""
        class="CVEditorTab__img"
        src="movie_2.png"
      />
    </div>
    <div>
      <div
        class="row"
      >
        <div
          class="form-group col-md-6 col-12"
        >
          <label
            class="form-label"
            for="startTime"
          >
            Od:
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                id="startTime"
                placeholder="Data rozpoczęcia"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="form-group col-md-6 col-12"
        >
          <label
            class="form-label"
            for="endTime"
          >
            Do:
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="form-control"
                id="endTime"
                placeholder="Data zakończenia"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="actionPlace"
        >
          Miejsce:
        </label>
        <input
          class="form-control"
          id="actionPlace"
          placeholder="Nazwa szkoły/miejsca pracy"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="actionDescription"
        >
          Opis:
        </label>
        <input
          class="form-control"
          id="actionDescription"
          placeholder="Profil/stanowisko ..."
          type="text"
          value=""
        />
      </div>
      <button
        class="itemsList__addButton btn btn-success"
        type="button"
      >
        + Dodaj
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <button
          class="form_navigation_prev btn btn-primary btn-block"
          disabled=""
          type="button"
        >
          ← Wstecz
        </button>
      </div>
      <div
        class="col-md-6 col-12"
      >
        <button
          class="btn btn-success btn-block"
          id="saveButton"
          type="submit"
        >
          Generuj CV
        </button>
      </div>
    </div>
  </div>
</div>
`;
